scripts/py_matter_yamltests/matter_yamltests/yaml_loader.py

```diff
@@ -124,7 +124,7 @@ def __check_test_step(self, content):
             response = content.get('response')
             if isinstance(response, list):
                 [self.__check_test_step_response(x) for x in response]
-            else:
+            elif isinstance(response, dict):
```

Comment:
how about if it is a str? Should there be an else branch? Or if not, a comment explaining why in case of a variable we do not need to do anything here.

Reply:
I have added a check for it that also allows to check the variable content.